### ISSUE TITLE
PubLookup.lookup to return an array of public keys

### ISSUE THREAD

**Issue by tomholub** [FlowCrypt/flowcrypt-ios#337]
after #553 

we are blindly choosing `.first` public key which is not ideal, and instead in the future we should return `[Contact]`. If we are pulling keys during compose to save them in db, save all of them

_Originally posted by @tomholub in https://github.com/FlowCrypt/flowcrypt-ios/issues/296#issuecomment-855239434_

**Comment by tomholub** [comment]
This is important for future work on key lifecycle functionality (rotation, expiration / extension, revocation)
